packages/connect/test/orders_channel_factory_test.ts

```diff
@@ -0,0 +1,33 @@
+import * as chai from 'chai';
+import * as dirtyChai from 'dirty-chai';
+import * as _ from 'lodash';
+import 'mocha';
+
+import { ordersChannelFactory } from '../src/orders_channel_factory';
+
+chai.config.includeStack = true;
+chai.use(dirtyChai);
+const expect = chai.expect;
+const emptyOrdersChannelHandler = {
+    onUpdate: _.noop.bind(_),
+    onError: _.noop.bind(_),
+    onClose: _.noop.bind(_),
+};
+
+describe('ordersChannelFactory', () => {
+    const websocketUrl = 'ws://localhost:8080';
+    describe('#createWebSocketOrdersChannelAsync', () => {
+        it('throws when input is not a url', () => {
+            const badUrlInput = 54;
+            expect(
+                ordersChannelFactory.createWebSocketOrdersChannelAsync(badUrlInput as any, emptyOrdersChannelHandler),
+            ).to.be.rejected();
+        });
+        it('throws when handler has the incorrect members', () => {
+            const badHandlerInput = {};
+            expect(
+                ordersChannelFactory.createWebSocketOrdersChannelAsync(websocketUrl, badHandlerInput as any),
+            ).to.be.rejected();
+        });
```

Comment:
Can we add some success cases?

Reply:
I tried setting up a quick test for this. It's actually harder than it looks because instantiating with the factory means that internally the client will try to connect to the web socket. We are able to do this in the actual channel tests because we can access the client and stub the `readyState` property. We cannot do this in the factory test.